include/libtorrent/torrent.hpp

```diff
@@ -1481,7 +1481,7 @@ namespace libtorrent
 // ----
 
 		// the maximum number of uploads for this torrent
-		unsigned int m_max_uploads:24;
+		std::uint32_t m_max_uploads:24;
```

Comment:
the adjacent fields in this bitfield need to have the same type too. Otherwise they will be separate "objects"/members.